### PR TITLE
Chapter 5: Generics

### PR DESCRIPTION
**Pull request by alexstyl**
#### Description

This PR looks into the usage of generics according to Chapter 5 of the Effective Java book and improves the usage of generics overall around the app. 

More specifically:
- Preferred the usage of Generics for references instead of concrete implementations (Item 26)
- Removed `"unchecked warnings` (Item 24)
##### Test(s) added

nope. Just updated the already existing ones

### CODE REVIEW

File: mobile/src/main/java/com/alexstyl/specialdates/addevent/WriteableAccountsProvider.java [hunk 1]
```diff
@@ -44,15 +44,16 @@ public WriteableAccountsProvider(AccountManager accountManager, PackageManager p
     }
 
     private static boolean accountIsWritable(String accountType) {
-        // TODO how do you even?
         return GOOGLE_ACCOUNT.equals(accountType);
     }
 
-    private static AuthenticatorDescription getAuthenticatorDescription(String type,
```
Comment (alexstyl) [inline mobile/src/main/java/com/alexstyl/specialdates/addevent/WriteableAccountsProvider.java]: There is no particular usage of Generics in here. I just simplified the for loop as the IDE suggested

**Comment by blundell** [comment]
> Preferred the usage of Generics for references instead of concrete implementations

https://github.com/alexstyl/Memento-Namedays/pull/61/commits/b647d90a45787c8185c8e36b3618c51378ad370a FYI `Even moar generic usage` this is not use of generics, this is 'coding to the interface' https://en.wikipedia.org/wiki/Liskov_substitution_principle 

is this what you meant to change? Maybe a bit more explanation in your PR description for what you are trying to achieve

**Comment by alexstyl** [comment]
@blundell Sounds the same. That specific commit is the Item 26 of the book as written on the description, isn't it?

File: mobile/src/main/java/com/alexstyl/specialdates/events/namedays/NamedayDatabaseRefresher.java [hunk 2]
```diff
@@ -31,8 +31,7 @@
 public class NamedayDatabaseRefresher {
 
     private static final Optional<Contact> NO_CONTACT = Optional.absent();
-    @SuppressWarnings("unchecked")
-    private static final List<ContactEvent> NO_EVENTS = Collections.EMPTY_LIST;
+    private static final List<ContactEvent> NO_EVENTS = Collections.emptyList();
```
Comment (blundell) [inline mobile/src/main/java/com/alexstyl/specialdates/events/namedays/NamedayDatabaseRefresher.java]: 👍

File: mobile/src/main/java/com/alexstyl/specialdates/upcoming/SettingsMonitor.java [hunk 3]
```diff
@@ -13,7 +13,7 @@
     private List<EventUpdatedMonitor> monitors;
 
     public static SettingsMonitor newInstance(Context context) {
-        ArrayList<EventUpdatedMonitor> monitors = new ArrayList<>(2);
+        List<EventUpdatedMonitor> monitors = new ArrayList<>(2);
```
Comment (blundell) [inline mobile/src/main/java/com/alexstyl/specialdates/upcoming/SettingsMonitor.java]: dat perf `2`

**Comment by blundell** [comment]
@alexstyl yeah i see "prefer the more generic type" 💯

**Comment by blundell** [comment]
i dont have admin to merge. lgtm!

**Pull request merged**